s3transfer/download.py

```diff
@@ -171,7 +172,8 @@ def _get_temp_fileobj(self):
         return f
 
     def _get_fileobj(self, filename):
-        f = self._osutil.open(filename, 'wb')
+        f = DeferredOpenFile(filename, mode='wb')
+        f.OPEN_METHOD = self._osutil.open
```

Comment:
We're already adding a `mode` argument, why not add an `open_function` argument as well?